target/linux/ramips/mt7621/base-files/etc/board.d/02_network

```diff
@@ -202,7 +203,8 @@ ramips_setup_macs()
 		wan_mac=$(mtd_get_mac_ascii factory wanmac)
 		label_mac=$(mtd_get_mac_binary radio 0x4)
 		;;
-	edimax,rg21s)
+	edimax,rg21s)|\
+	edimax,ra21s)
 		lan_mac=$(mtd_get_mac_ascii u-boot-env ethaddr)
 		wan_mac=$(mtd_get_mac_ascii u-boot-env wanaddr)
```

Comment:
Does the device have a label and what's the address on it (to which device does it correspond?):
https://openwrt.org/docs/guide-developer/mac.address#label_mac_address

Reply:
Is this change absolutely necessary? I am not so familiar with the codebase and not sure that ready for major changes. RA21S and RG21S are almost the same devices and almost all changes was copied from that device.

Reply:
I'm not sure what major change you are referring to here. I just wanted to know whether the device has a label MAC address and which interface it refers to. If you provide that information, I will assist you with putting it at the right place in case you are unsure.
Despite, note that this particular change here has wrong syntax and would need alphabetic sorting.

Reply:
Device has a sticker with WLAN ports information. Something like 

> 2.5GGhz: 74DAXXXXXX01
> PIN CODE: 12345
> SSID: edimax.setup.1f

and 
> 5GGhz: 74DAXXXXXX02
> PIN CODE: 56789
> SSID: edimax.setup.5G

If I uderstand correctly, I have to init `label_mac` variable in the `02_network` file. E.g. by adding  `label_mac=$wan_mac` line. However device I does not have sticker with WAN MAC.

Want would be best solution about the `label_mac`  here?

Reply:
You could just use the first address on the sticker (looks like it's 2.5 GHz in your case).
In the DTS, the caldata for this WiFi is read from <&factory 0x0>. As you use ralink,mtd-eeprom there, this WiFi will automatically get it's address from offset +4 of this location (unless you set it up wrong and the normal WiFi address is not correct).
So, in the 02_network file you can just initialize the label MAC address from that very same location:
`label_mac=$(mtd_get_mac_binary factory 0x4)`

Reply:
yes, `factory + 0x4` contains MAC for the 2.5 GHz device.  
I pushed `label_mac=$(mtd_get_mac_binary factory 0x4)`